azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -700,27 +700,35 @@ def _update_client_secret_to_extension(
         properties = aio_extension["properties"]
 
         config_settings: dict = properties.get("configurationSettings", {})
-        if not config_settings:
-            properties["configurationSettings"] = {}
 
-        config_settings["connectors.values.securityPki.applicationCert"] = application_cert
-        config_settings["connectors.values.securityPki.subjectName"] = subject_name
-        config_settings["connectors.values.securityPki.applicationUri"] = application_uri
+        desired_config_settings = config_settings.copy()
+        desired_config_settings["connectors.values.securityPki.applicationCert"] = application_cert
+        desired_config_settings["connectors.values.securityPki.subjectName"] = subject_name
+        desired_config_settings["connectors.values.securityPki.applicationUri"] = application_uri
 
-        aio_extension["properties"]["configurationSettings"] = config_settings
-
-        status_text = (
-            f"Updating IoT Operations extension to use {application_cert}..."
-            if application_cert
-            else "Rollback client certificate from IoT Operations extension..."
+        delta = calculate_config_delta(
+            current=config_settings,
+            target=desired_config_settings,
         )
 
-        with console.status(status_text):
-            return self.resource_map.connected_cluster.update_aio_extension(
-                extension_name=aio_extension["name"],
-                properties=properties,
+        if delta:
+            status_text = (
+                f"Updating IoT Operations extension to use {application_cert}..."
+                if application_cert
+                else "Rollback client certificate from IoT Operations extension..."
```

Comment:
How is this rollback supposed to work?  application_cert is optional and when None the opcua connector rollsback to a self-signed cert? Can you please describe the flow?

Reply:
yeah, rollback is basically setting the 3 properties to "", therefore when application_cert is "", we show status text as rollback

Reply:
also application_cert is required not optional for this function

Reply:
What happens if the other properties are not ""? Is that a possible condition? The status text just checks application_cert.

Reply:
talked offline